src/main/java/org/opensearch/security/privileges/SecurityIndexAccessEvaluator.java

```diff
@@ -149,6 +151,14 @@ public PrivilegesEvaluatorResponse evaluate(final ActionRequest request, final T
     }
 
     private boolean matchAnySystemIndices(final Resolved requestedResolved){
-        return systemIndexEnabled && systemIndexMatcher.matchAny(requestedResolved.getAllIndices());
+        return !getProtectedIndexes(requestedResolved).isEmpty();
+    }
+
+    private List<String> getProtectedIndexes(final Resolved requestedResolved) {
+        final List<String> protectedIndexes = requestedResolved.getAllIndices().stream().filter(securityIndex::equals).collect(Collectors.toList());
```

Comment:
I'm slightly confused. Won't this filter only match `securityIndex` and collect them. I'm not quite sure how is it matching against all protected indices? Or is the goal to match against securityIndex only

Reply:
The security index should always be checked, this statement will create a collection with 0 items, or 1 item being the security index, if it was found.  

In the following line, if there are system indexes (not the security index), those will be added to the results in the branch statement, so long as the configuration switch `systemIndexEnabled` is true.